### ISSUE TITLE
Documentation for trusted_hosts

### ISSUE THREAD

**Issue by AntonOfTheWoods** [pdm-project/pdm#1306]
## Is your feature request related to a problem? Please describe.

I just added 

```
pdm config --local pypi.url http://mirrors.aliyun.com/pypi/simple/
```

And tried to add a package. I got a message:

```
[CandidateNotFound]: Unable to find candidates for requests. There may exist some issues with the package name or network condition.
```

And in the log

```
unearth: Collecting links from http://mirrors.aliyun.com/pypi/simple/requests/
unearth: Skip http://mirrors.aliyun.com/pypi/simple/requests/ for not being trusted, please add it to `trusted_hosts` list
```

I then searched and there is absolutely no documentation on where `trusted_hosts` is configured. In fact, I still don't know how to configure it. In the code, I saw:

```
if host not in trusted_hosts and not source.get("verify_ssl", True):
```

So I thought "maybe the issue will go away if I try with https?". It did, but it would be good to have this documented somewhere!

## Describe the solution you'd like

It would be good to have some more documentation on how to configure `trusted_hosts` and the gotcha with http. I just found out by guessing so can't contribute the docs unfortunately!

**Comment by frostming** [comment]
`pdm config --local pypi.verify_ssl true`, it is documented https://pdm.fming.dev/latest/usage/configuration/

By the way why are you not using https://mirrors.aliyun.com/pypi/simple/ ? The mirror supports HTTPS

**Comment by AntonOfTheWoods** [comment]
Lots of docs *don't* use HTTPS, so the law of copy/paste says by default a normal person won't do it by default. I assumed it would support https so I tried that, and it worked. That is not a very good developer experience though.

I realise that there are possibly a few ways around this but no normal person will think that "verify_ssl" is for allowing `http` to be used instead of `https`. The only normal assumption is to ignore SSL errors if you get one on an `https` link.

But the issue is not that, the issue is that there is no documentation. The message says nothing about configuring `verify_ssl`, it talks about `trusted_hosts`. So I search for documentation on how to configure `trusted_hosts`, and there is none.

**Comment by frostming** [comment]
`trusted_hosts` is thrown by `unearth`, but PDM is using the term `verify_ssl`.

**Comment by AntonOfTheWoods** [comment]
As a new dev coming to use your software, you would prefer them to have to learn about all your dependencies and go through your code, rather than have some documentation for the error messages that the developers see when they use your application?

**Comment by frostming** [comment]
Firstly, the mirror should redirect HTTP to HTTPS but it doesn't. Secondly, the docs shouldn't recommend HTTP if the mirror supports HTTPS, and you should realize the docs are outdated.

**Comment by AntonOfTheWoods** [comment]
I thought using pdm would be easier now I am in the mainland but... 算了吧

**Comment by frostming** [comment]
No defending but as an OSS user, you shouldn't demand the maintainer to do something for YOU and shit on them if they don't. If you think the documentation sucks just send a PR and we'll be happy to review it. I have explained to you what the current term is being used and now you have been ready to start a contribution. But it is okay if you don't want to and leave PDM.

**Comment by AntonOfTheWoods** [comment]
Dude, I have dedicated the last 3 years of my life entirely to open source, and know very, very well the pain of not having useful contributions.

It is very difficult when the maintainer appears to deny there is an issue and blames *other* people. All I was suggesting was that it wasn't clear to a newcomer and you were suggesting everyone else was to blame so there is no problem.

And I am still not clear on what the situation is. Are you saying that `trusted_hosts` is *not* a value that can be set by pdm? That the only config option is `verify_ssl`？

When the issue is obvious and I can usefully submit a PR, I will immediately do that, as I did with the asdf issue. When I don't know how something actually works, how can I tell others about it?
